src/Smalot/PdfParser/Document.php

```diff
@@ -144,9 +149,97 @@ protected function buildDetails()
             $details['Pages'] = 0;
         }
 
+        $details = array_merge($details, $this->metadata);
+
         $this->details = $details;
     }
 
+    /**
+     * Extract XMP Metadata
+     */
+    public function extractXMPMetadata(string $content): void
+    {
+        $xml = xml_parser_create();
+        xml_parser_set_option($xml, \XML_OPTION_SKIP_WHITE, 1);
+
+        if (xml_parse_into_struct($xml, $content, $values, $index)) {
+
+            $metadata = [];
+            $stack = [];
+            foreach ($values as $val) {
+
+                // Standardize to lowercase
+                $val['tag'] = strtolower($val['tag']);
+
+                // Ignore structural x: and rdf: XML elements
+                if (strpos($val['tag'], 'x:') === 0) continue;
+                if (strpos($val['tag'], 'rdf:') === 0 && 'rdf:li' != $val['tag']) continue;
+  
+                switch ($val['type']) {
+                    case 'open':
+                        // Create an array of list items
+                        if ('rdf:li' == $val['tag']) {
+                            $metadata[] = [];
+
+                            // Move up one level in the stack
+                            $stack[count($stack)] = &$metadata;
+                            $metadata = &$metadata[count($metadata) - 1];
```

Comment:
Can you explain this and the next lines when using `&$metadata` with one or two sentences? 

I'd rather use a counter instead of `count` above, but its just a minor thing. The usage of `&$metadata` is a bit strange.

Reply:
Sure. The output of xml_parse_into_struct is a single dimensional array, and the $stack is a last-on, first-off array of pointers to positions in $metadata, while iterating through it, that potentially turn the results into a more intuitive multi-dimensional array. When an "open" XML tag is encountered, we save the current $metadata context in the $stack, then create a child array of $metadata and make that the *current* $metadata context. When a "close" XML tag is encountered, the operations are reversed: the most recently added $metadata context from $stack (IOW, the *parent* of the current element) is set as the current $metadata context.